frontend-plugin-core/src/test/java/com/github/eirslett/maven/plugins/frontend/lib/UtilsTest.java

```diff
@@ -0,0 +1,67 @@
+package com.github.eirslett.maven.plugins.frontend.lib;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class UtilsTest {
+
+  @Test
+  public void testImplode() {
+    final String separator = "Bar";
+    final List<String> elements = new ArrayList<String>();
+
+    Assert.assertEquals("", Utils.implode(separator, elements));
+
+    elements.add("foo");
+    elements.add("bar");
+    Assert.assertEquals("foo bar", Utils.implode(separator, elements));
+  }
+
+  @Test
+  public void testIsRelative() {
+    Assert.assertTrue(Utils.isRelative("foo/bar"));
+
+    Assert.assertFalse(Utils.isRelative("/foo/bar"));
+    Assert.assertFalse(Utils.isRelative("file:foo/bar"));
+    Assert.assertFalse(Utils.isRelative("C:\\SYSTEM"));
+  }
+
+  @Test
+  public void testMerge() {
+    final List<String> first = new ArrayList<String>();
+    first.add(null);
+    final List<String> second = new ArrayList<String>();
+    second.add("a/b/c");
+    final List<String> actual = Utils.merge(first, second);
+
+    final List<String> arrayList = new ArrayList<String>();
+    arrayList.add(null);
+    arrayList.add("a/b/c");
+
+    Assert.assertEquals(arrayList, actual);
+  }
+
+  @Test
+  public void testNormalize() {
+    final String path = "/foo/bar/baz";
+
+    Assert.assertEquals("/foo/bar/baz", Utils.normalize(path));
```

Comment:
I don't think the logic is being correctly tested here.  There has to be a way to mock or set `file.separator` so that something is actually replaced in the string.

Reply:
Agreed, this test has been removed.
**fs = File.separator** is one way around it but this involves changing values in the test. As discussed, mocking appears to be the right way to go to ensure consistent results on all platforms but, following a chat with Joel, this looks to be non-trivial. A ticket will be logged to address this sort of situation in Diffblue Cover.